File: app/src/main/java/com/lucciola/haskellanywhereforandroid/uiaction/HaskellAction.java
package com.lucciola.haskellanywhereforandroid.uiaction;

import com.lucciola.haskell.Haskell;
import com.lucciola.haskellanywhereforandroid.View;
import com.lucciola.haskellanywhereforandroid.ProgramManager;

/**
 * Created by hiden on 2017/06/17.
 */

public class HaskellAction extends UIAction {
    private Haskell haskell;

    public HaskellAction(ProgramManager arg0, View arg2) {
        super(arg0, arg2);
        this.haskell = new Haskell();
    }

    @Override
    protected void onPreExecute() {
        this.view.scrollDown();
        this.view.showDialog("Please wait...", "Runnnig the Haskell program...");
    }

    @Override
    protected String doInBackground(Void... params) {
        haskell.setProgram(this.manager.getProgram());
        return haskell.run().getMessage() + "\n";
    }

    @Override
    protected void onPostExecute(String arg0) {
        this.view.addMonitorText(this.manager.getCurrentInput() + arg0);
        this.view.hideDialog();
        this.view.scrollDown();
    }
}

File: app/src/main/java/com/lucciola/haskellanywhereforandroidkt/uiaction/HaskellAction.kt
package com.lucciola.haskellanywhereforandroidkt.uiaction

import com.lucciola.haskell.Haskell
import com.lucciola.haskellanywhereforandroidkt.View
import com.lucciola.haskellanywhereforandroidkt.ProgramManager

/**
 * Created by hiden on 2017/06/17.
 */

class HaskellAction(arg0: ProgramManager, arg2: View) : UIAction(arg0, arg2) {
    private val haskell: Haskell

    init {
        this.haskell = Haskell()
    }

    override fun onPreExecute() {
        this.view.scrollDown()
        this.view.showDialog("Please wait...", "Runnnig the Haskell program...")
    }

    override fun doInBackground(vararg params: Void): String {
        haskell.setProgram(this.manager.program)
        return haskell.run().message + "\n"
    }

    override fun onPostExecute(arg0: String) {
        this.view.addMonitorText(this.manager.currentInput!! + arg0)
        this.view.hideDialog()
        this.view.scrollDown()
    }
}
